### PR TITLE
autoinstall_templates are installed into /var/lib/cobbler/templates

### PR DESCRIPTION
**Pull request by opoplawski**
This isn't tested, but the use of autoinstall_templates seems wrong

### CODE REVIEW

**Comment by SchoolGuy** [comment]
I will try to cross-check all locations about both paths and then have a look if this fixes the problem.

**Review by SchoolGuy — CHANGES_REQUESTED** [review]
This is a good catch and I would love to have it but I think two minor details need to be changed before we can merge this.

File: tests/test_data/settings_old
Comment (watologo1) [reply tests/test_data/settings_old:95]: @SchoolGuy Are you sure? It could still get adjusted, it was using the old "/var/lib/cobbler/autoinstall_templates" as well. So it should not hurt and looks cleaner to me?

**Review by watologo1 — APPROVED** [review]
Not sure autoyast/kickstart templates still do exists in /srv/www/... at all anymore.
They should be generated at runtime when calling:
http://cobbler.xy.de/cblr/svc/op/autoinstall/system/foo.xy.de
But it's still better... ;)
Looks good to me.

File: tests/test_data/settings_old
Comment (SchoolGuy) [reply tests/test_data/settings_old:95]: Yes I am. This is for for migration from an old settings file to a new one. This should never be adjusted to be able to simulate an old file, updating this could possibly cause incorrect behavior of the upgrade tests.

**Review by nodeg — APPROVED** [review]
From my point of view this seems legit. After examining the folder structure in my cobbler VM, there is only a `templates` folder under `/var/lib/cobbler`.

```bash
$ ll /var/lib/cobbler          
total 84K
drwxr-xr-x 1 root   root  114 2021-02-18 11:51 collections
-rw-r--r-- 1 root   root  79K 2021-03-04 13:09 distro_signatures.json
drwxr-xr-x 1 root   root   58 2021-03-30 09:20 grub_config
drwxr-xr-x 1 root   root    0 2021-02-18 11:51 loaders
-rw-r--r-- 1 root   root    0 2021-02-18 11:55 lock
drwxr-xr-x 1 root   root  140 2021-03-30 09:20 scripts
drwxr-xr-x 1 root   root 1.6K 2021-03-30 09:20 snippets
drwxr-xr-x 1 root   root  424 2021-03-30 09:20 templates
drwxr-xr-x 1 root   root   60 2021-02-18 11:51 triggers
-rw-r--r-- 1 wwwrun root 1.0K 2021-03-30 09:21 web.ss
drwxr-xr-x 1 wwwrun root    0 2021-02-18 11:51 webui_sessions

$ ll /var/lib/cobbler/templates
total 64K
-rw-r--r-- 1 root root  115 2021-02-18 11:51 default.ks
-rw-r--r-- 1 root root   22 2021-02-18 11:51 esxi4-ks.cfg
-rw-r--r-- 1 root root   22 2021-02-18 11:51 esxi5-ks.cfg
drwxr-xr-x 1 root root    0 2021-02-18 11:51 install_profiles
-rw-r--r-- 1 root root 1.4K 2021-02-18 11:51 legacy.ks
-rw-r--r-- 1 root root  669 2021-02-18 11:51 powerkvm.ks
-rw-r--r-- 1 root root  292 2021-02-18 11:51 pxerescue.ks
-rw-r--r-- 1 root root 2.8K 2021-02-18 11:51 sample_autoyast.xml
-rw-r--r-- 1 root root  328 2021-02-18 11:51 sample_esxi4.ks
-rw-r--r-- 1 root root  390 2021-02-18 11:51 sample_esxi5.ks
-rw-r--r-- 1 root root  390 2021-02-18 11:51 sample_esxi6.ks
-rw-r--r-- 1 root root  390 2021-02-18 11:51 sample_esxi7.ks
-rw-r--r-- 1 root root 1.7K 2021-02-18 11:51 sample.ks
-rw-r--r-- 1 root root 1.8K 2021-02-18 11:51 sample_legacy.ks
-rw-r--r-- 1 root root 3.4K 2021-02-18 11:51 sample_old.seed
-rw-r--r-- 1 root root 6.6K 2021-02-18 11:51 sample.seed
```

I tested the following with a Debian 10 image
> If no autoinstall template is specified to profile add, use this template.

```bash
$ cobbler profile report
Name                           : debian-x86_64
Automatic Installation Template : sample.seed
Automatic Installation Metadata : {}
(...)
```
The `sample.seed` file is only found under `/var/lib/cobbler/templates`.

Furthermore when you look e.g. at the `settings.py` you find the following

```python
"autoinstall_templates_dir": ["/var/lib/cobbler/templates", "str"],
```
and the `setup.py` mentions the following
```python
data_files=[
(...)
            ("%s/templates" % libpath, glob("autoinstall_templates/*")),
            ("%s/templates/install_profiles" % libpath, glob("autoinstall_templates/install_profiles/*")),
(...)
```

The only thing which should not be touched as @SchoolGuy already mentioned is the `settings_old` file. Otherwise LGTM.

**Comment by dansou901** [comment]
While you're at it, may I remind you that we should try to move all folders where stuff gets created in cobbler which should be persisted in a docker volume when using a container in one place so volume management is not so messy as it is now?

**Review by SchoolGuy — APPROVED** [review]
LGTM if CI passes

**Pull request merged**
